mesonbuild/interpreter/dependencyfallbacks.py

```diff
@@ -340,6 +340,18 @@ def lookup(self, kwargs: TYPE_nkwargs, force_fallback: bool = False) -> Dependen
                         self._subproject_impl(subp_name, varname)
                     break
 
+                # Check if we could have a matching Rust crate imported from
+                # Cargo.lock, with expected api version. We know there is no
+                # wrap-file defined to match this crate.
+                if name.endswith('-rs') and required:
```

Comment:
This is a terrible heuristic. What is stopping me from name a C subproject with a -rs suffix. cc @xclaesse

Reply:
In case we don't identify a matching cargo wrap, we just continue to next dependency, so the heuristic can't create a false positive.

The only possibility is if someone has a cargo.lock with this exact dependency, and needs a C library named exactly with -rs suffix, which is, IMHO, unlikely to be a real use case.

Reply:
This ties into what I said before, I think.

Reply:
https://github.com/mesonbuild/meson/pull/13448#discussion_r1714344951